### PR TITLE
regtests: install main branch commit of `stcal` and `stdatamodels`

### PR DESCRIPTION
**Pull request by zacharyburnett**
This PR installs the latest commits of `stcal`, `stdatamodels`, ~~and `stpipe`~~ in the Jenkins regression tests

**Checklist for maintainers**
- [ ] ~~added entry in `CHANGES.rst` within the relevant release section~~
- [ ] updated or added relevant tests
- [ ] updated relevant documentation
- [ ] added relevant milestone
- [x] added relevant label(s)
- [x] ran regression tests, post a link to the Jenkins job below.
      [How to run regression tests on a PR](https://github.com/spacetelescope/jwst/wiki/Running-Regression-Tests-Against-PR-Branches)
- [ ] Make sure the JIRA ticket is [resolved properly](https://github.com/spacetelescope/jwst/wiki/How-to-resolve-JIRA-issues)

### CODE REVIEW

File: JenkinsfileRT [hunk 1]
```diff
@@ -78,6 +78,7 @@ bc0.pip_reqs_files = ['requirements-sdp.txt']
 bc0.build_cmds = [
     "pip install certifi -U --force-reinstall",
     "pip install -e .[test,sdp] --no-cache-dir",
+    "pip install git+https://github.com/spacetelescope/stdatamodels.git@master git+https://github.com/spacetelescope/stpipe.git@master git+https://github.com/spacetelescope/stcal.git@main",
```
Comment (hbushouse) [inline JenkinsfileRT:81]: I think we can leave `stpipe` out of this list for the RT build.

File: JenkinsfileRT_dev [hunk 2]
```diff
@@ -78,6 +78,7 @@ bc0.pip_reqs_files = ['requirements-dev.txt']
 bc0.build_cmds = [
     "pip install certifi -U --force-reinstall",
     "pip install -e .[test,sdp] --no-cache-dir",
+    "pip install git+https://github.com/spacetelescope/stdatamodels.git@master git+https://github.com/spacetelescope/stpipe.git@master git+https://github.com/spacetelescope/stcal.git@main",
```
[the remark this answers is not in the document]
Comment (zacharyburnett) [reply JenkinsfileRT_dev:81]: oh, good point

**Comment by zacharyburnett** [comment]
new regtest run is at https://plwishmaster.stsci.edu:8081/blue/organizations/jenkins/RT%2FJWST-Developers-Pull-Requests/detail/JWST-Developers-Pull-Requests/611/pipeline

**Review by hbushouse — APPROVED** [review]

File: JenkinsfileRT_dev [hunk 3]
```diff
@@ -95,9 +95,19 @@ bc0.test_configs = [data_config]
 bc1 = utils.copy(bc0)
 bc1.nodetype = 'macos'
 bc1.name = 'macos-unstable-deps'
-bc1.build_cmds = ["pip install -e ."]
+bc0.pip_reqs_files = ['requirements-dev.txt']
```
Comment (hbushouse) [inline JenkinsfileRT_dev:98]: Just noticed this now, after the fact. Should this be `bc1` here, as opposed to `bc0`?  Looks like a copy-n-paste error?

Comment (zacharyburnett) [reply JenkinsfileRT_dev:98]: good catch; https://github.com/spacetelescope/jwst/pull/7504

**Pull request merged**